oauth2_provider/oauth2_validators.py

```diff
@@ -760,7 +788,13 @@ def get_oidc_claims(self, token, token_handler, request):
         claims = {}
 
         for k, v in data.items():
-            claims[k] = v(request) if callable(v) else v
+            if not self.oidc_claim_scope or (
+                token
+                and hasattr(token, "scopes")
```

Comment:
I'm not super-happy about having to test for token and token.scopes. It's not clear if this is a real possibility or if the test cases aren't providing enough of a dummy token.